### PR TITLE
Refactor LDAP  and CLI product tests

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

**Comment by sanjay990** [comment]
@anusudarsan @amrutagokhale

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliUtils.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import java.io.IOException;
+import java.util.List;
+
+import static com.google.common.io.Resources.getResource;
+import static com.google.common.io.Resources.readLines;
+import static com.teradata.tempto.process.JavaProcessLauncher.defaultJavaProcessLauncher;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+
+public class PrestoCliUtils extends ProductTest
```
Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliUtils.java]: This class doesn't actually have all the characteristics of a Utility class in Java (other classes need to extend this class). So rather rename this as something other than `PrestoCliUtils` such as `PrestoCliLauncher`.

Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliUtils.java]: Also, move this to a new line: `extends ProductTest`.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliLauncher.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @Named("databases.presto.host")
+    protected String serverHost;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    protected String serverAddress;
+
+    protected static PrestoCliProcess presto;
+
+    public PrestoCliLauncher()
```
Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliLauncher.java]: Can you make this `protected` too?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliLauncher.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Inject
+    @Named("databases.presto.host")
+    protected String serverHost;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    protected String serverAddress;
+
+    protected static PrestoCliProcess presto;
```
Comment (amrutagokhale) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliLauncher.java]: From what we discussed, I thought we aren't making this `static`?

Comment (sanjay990) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliLauncher.java]: yes, will fix that

**Comment by amrutagokhale** [comment]
LGTM.

**Comment by sanjay990** [comment]
Merged

**Pull request closed without merging**